pkg/agent/ipassigner/ip_assigner_linux_test.go

```diff
@@ -0,0 +1,947 @@
+// Copyright 2024 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ipassigner
+
+import (
+	"fmt"
+	"net"
+	"reflect"
+	"sync"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/vishvananda/netlink"
+	gomock "go.uber.org/mock/gomock"
+	"k8s.io/apimachinery/pkg/util/sets"
+
+	respondertest "antrea.io/antrea/pkg/agent/ipassigner/responder/testing"
+	crdv1b1 "antrea.io/antrea/pkg/apis/crd/v1beta1"
+)
+
+func netlinkLinkSetup(link netlink.Link) error {
+	return nil
+}
+
+func ensureRPFInt(name string, y int) error {
+	return nil
+}
+
+func DummyInterfaceByName(name string) (*net.Interface, error) {
+	return &net.Interface{
+		Index:        0,
+		MTU:          1500,
+		Name:         "eth0",
+		HardwareAddr: []byte{0x00, 0x11, 0x22, 0x33, 0x44, 0x55},
+	}, nil
+
+}
+
+func netLinkAddel(vlan netlink.Link) error {
+	return nil
+}
+
+type AddrRecord struct {
+	Link    netlink.Link
+	Address []*netlink.Addr
+}
+
+var addedAddresses []*AddrRecord
+
+func newFakeNetworkInterface() *net.Interface {
+	return &net.Interface{
+		Index:        0,
+		MTU:          1500,
+		Name:         "eth0",
+		HardwareAddr: []byte{0x00, 0x11, 0x22, 0x33, 0x44, 0x55},
+	}
+}
+
+type dummyDeviceMock struct {
+	addedIPs []net.IPNet
+	vlanID   int
+}
+
+func (d *dummyDeviceMock) Attrs() *netlink.LinkAttrs {
+	if d.vlanID == 0 {
+		return &netlink.LinkAttrs{Name: "antrea-dummy0"}
+	}
+	return &netlink.LinkAttrs{Name: fmt.Sprintf("antrea-ext.%d", d.vlanID)}
+}
+
+func (d *dummyDeviceMock) AddrAdd(addr *netlink.Addr) error {
+	ipNet := net.IPNet{
+		IP:   addr.IPNet.IP,
+		Mask: net.CIDRMask(32, 32),
+	}
+	d.addedIPs = append(d.addedIPs, ipNet)
+	return nil
+}
+
+func (d *dummyDeviceMock) Type() string {
+	return "dummy"
+}
+
+func DummyDeviceMockVlan(vlanID int) *dummyDeviceMock {
+	return &dummyDeviceMock{vlanID: vlanID}
+}
+
+func addrAddDel(link netlink.Link, addr *netlink.Addr) error {
+
+	fmt.Println("calling addrAddDel")
```

Comment:
remove debug print

But I think it doesn't need this at all if it just replaces the netlink interface with a mock implementation